code-challenges.test.js

```diff
@@ -23,30 +39,86 @@ const secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
+const codedString = (str) => {
+    let code = {"a":4, "e":3, "i":1, "o":0}
+    return str.split("").map((value) =>{
+        return(object.keys(code).includes(value.toLowercase())) ? code[value.toLowercase()] : value
+    }).join("")
+}
+
+// pseudo code: 
+// input: function that takes in a string
+// output/return: a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
+// I created a function that took in a string and returned an array of strings with the split "" and used the .map to iterate through every element in the array. Then i ask for a return and used the .keys method which returns an array of the objects strings. i also used the .includes() method that looks for one string that could be in another string returning the true or false as it should. And the .toLowerCase converts a string to lower case letters
+
+
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+describe("letter", () => {
+    it("takes in an array of words and a single letter and returns an array of all the words containing that particular letter", () => {
+      expect(letter(filterletterA)).toEqual(["Mango", "Apricot", "Peach"])
+      expect(letter(filterletterB)).toEqual(["Cherry", "Blueberry", "Peach"])
+
+      const filterLetterA = "a"
+      // Expected output: ["Mango", "Apricot", "Peach"]
+      const filterLetterE = "e"
+      // Expected output: ["Cherry", "Blueberry", "Peach"]
+      const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+    })
+  })
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+// pseudo code: 
+// input: array of words and a single letter
+// output/return: array of all the words containing that particular letter.
+// The letter function takes an array of words and a single letter. It returns a new array that only includes the words from the original array that have the specified letter. The function compares the lowercase versions of the words and the letter to make the comparison case-insensitive. The array now has the words that have the specified letter.
 
 // b) Create the function that makes the test pass.
 
+function letter(words, letter) {
+    return words.filter(word => word.toLowerCase().includes(letter.toLowerCase()));
+  }
+
```

Comment:
Since your function takes in two parameters the function call on your expect statement should also have two arguments.
```js
expect(letter(fruitArray, filterletterA)).toEqual(["Mango", "Apricot", "Peach"])
expect(letter(fruitArray, filterletterE)).toEqual(["Cherry", "Blueberry", "Peach"])
```